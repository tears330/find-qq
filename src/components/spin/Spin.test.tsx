import { render } from '@testing-library/react';
import { Spin } from '.';
describe('Spin', () => {
  it('render spin', () => {
    const { baseElement, rerender } = render(
      <Spin spinning={false}>
        <>test</>
      </Spin>,
    );
    expect(baseElement).toMatchSnapshot();
    rerender(
      <Spin spinning={true}>
        <>test again</>
      </Spin>,
    );
    expect(baseElement).toMatchSnapshot();
    rerender(
      <Spin spinning={false}>
        <>test</>
      </Spin>,
    );
    expect(baseElement).toMatchSnapshot();
  });
});
