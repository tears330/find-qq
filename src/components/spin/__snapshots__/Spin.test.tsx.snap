// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Spin render spin 1`] = `
<body>
  <div>
    test
  </div>
</body>
`;

exports[`Spin render spin 2`] = `
<body>
  <div>
    <div
      class="com-spin"
    >
      <div
        class="com-spin-loading"
      >
        <div />
        <div />
        <div />
        <div />
        <div />
        <div />
        <div />
        <div />
      </div>
      <div
        class="com-spin-wrapper"
      >
        test again
      </div>
    </div>
  </div>
</body>
`;

exports[`Spin render spin 3`] = `
<body>
  <div>
    test
  </div>
</body>
`;
